src/services/refactors/convertToOptionalChainExpression.ts

```diff
@@ -182,12 +206,18 @@ namespace ts.refactor.convertToOptionalChainExpression {
     }
 
     /**
-     * Gets the right-most property access expression.
+     * Gets a property access expression which may be nested inside of a call expression or binary expression. The final
+     * expression in an && chain will occur as the right child of the parent binary expression, unless it is a call expression
+     * or is followed by a different binary operator.
+     * @param node the right child of a binary expression or a call expression.
      */
     function getLastPropertyAccessChain(node: Expression): PropertyAccessExpression | undefined {
+        // foo && |foo.bar === 1|; - here the right child of the && binary expression is another binary expression.
+        // the rightmost member of the && chain should be the leftmost child of that expression.
         if (isBinaryExpression(node)) {

```

Comment:
What would be an example we care to test, something like `a && (a).b`?

Reply:
Yep, `foo && (foo.bar === 1)` seems more realistic since the precedence between different binary operators can be hard to remember, but I think if you put `skipParentheses` in the right places, you can trivially handle `foo && (((foo).bar) === 1)` 😁